### PR TITLE
DAOS-6419 test: Add checksum feature tags

### PR DESCRIPTION
**Pull request by daltonbohning**
Added checksum feature tag to avocado tests using checksums.
Since some of the osa test files were modified, also updated all osa test tag formats.

Organized test tags on four lines:
- frequency
- test stage
- features
- tests

### CODE REVIEW

**Comment by daltonbohning** [comment]
Hardware large reported: ERROR: No tests or tags found via ['checksum,hw,large']

**Comment by daltonbohning** [comment]
All modified test cases are non-failing and the tags are being picked up with the "Features" commit pragma:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-4420/4/artifact/Functional/

File: src/tests/ftest/osa/osa_online_extend.py [hunk 1]
```diff
@@ -158,8 +158,10 @@ def test_osa_online_extend(self):
         """Test ID: DAOS-4751
         Test Description: Validate Online extend
 
-        :avocado: tags=all,pr,daily_regression,hw,medium,ib2
-        :avocado: tags=osa,osa_extend,online_extend
+        :avocado: tags=all,pr,daily_regression
+        :avocado: tags=hw,medium,ib2
+        :avocado: tags=osa
```
Comment (rpadma2) [inline src/tests/ftest/osa/osa_online_extend.py:163]: You can add checksum here.

File: src/tests/ftest/osa/osa_online_parallel_test.py [hunk 2]
```diff
@@ -207,7 +207,9 @@ def test_osa_online_parallel_test(self):
 
         Test Description: Runs multiple OSA commands/IO in parallel
 
-        :avocado: tags=all,pr,daily_regression,hw,medium,ib2,osa
+        :avocado: tags=all,pr,daily_regression
+        :avocado: tags=hw,medium,ib2
+        :avocado: tags=osa
```
Comment (rpadma2) [inline src/tests/ftest/osa/osa_online_parallel_test.py:212]: checksum

File: src/tests/ftest/osa/osa_online_reintegration.py [hunk 3]
```diff
@@ -171,8 +171,10 @@ def test_osa_online_reintegration(self):
 
         Test Description: Validate Online Reintegration
 
-        :avocado: tags=all,pr,daily_regression,hw,medium,ib2,osa
-        :avocado: tags=online_reintegration,DAOS_5610
+        :avocado: tags=all,pr,daily_regression
+        :avocado: tags=hw,medium,ib2
+        :avocado: tags=osa,DAOS_5610
```
Comment (rpadma2) [inline src/tests/ftest/osa/osa_online_reintegration.py:176]: checksum

**Comment by rpadma2** [comment]
Any idea why ior_small and mdtest_small is not included. I believe those tests are also run with checksum enabled.

**Comment by daltonbohning** [comment]
> Any idea why ior_small and mdtest_small is not included. I believe those tests are also run with checksum enabled.

I'm using a script to generate a list of tests that use checksums. These tests, and the osa tests pointed out, were not being tagged by the script. I have adjusted the script, and now all of these (and more) are tagged. I will need to incorporate these and re-push.

Thanks!

**Comment by daltonbohning** [comment]
Latest run shows that all tests are picked up, except for `full_regression`, which is the expected behavior of the `Features:` tag:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-4420/6/artifact/Functional/

**Review by rpadma2 — APPROVED** [review]

**Review by mjean308 — APPROVED** [review]
lgtm

**Review by tanabarr — APPROVED** [review]

**Pull request merged**